reference/architecture/overview.md

```diff
@@ -1,204 +1,145 @@
-machine.
+**Main task**: Provides {{site.prodname}} networking for Kubernetes clusters. 
 
-etcd is responsible for performing the following tasks:
+The {{site.prodname}} binary that presents this API to Kubernetes is called the CNI plugin, and must be installed on every node in the Kubernetes cluster. The {{site.prodname}} CNI plugin allows you to use {{site.prodname}} networking for any orchestrator that makes use of the CNI networking specification. Configured through the standard {% include open-new-window.html text='CNI configuration mechanism' 
+url='https://github.com/containernetworking/cni/blob/master/SPEC.md#network-configuration' %}, and [{{site.prodname}} CNI plugin]({{site.baseurl}}/reference/cni-plugin/configuration).
 
-#### Data storage
 
-etcd stores the data for the {{site.prodname}} network in a distributed,
-consistent, fault-tolerant manner (for cluster sizes of at least three
-etcd nodes). This set of properties ensures that the {{site.prodname}} network is
-always in a known-good state, while allowing for some number of the
-machines hosting etcd to fail or become unreachable.
+### Datastore plugin
 
-This distributed storage of {{site.prodname}} data also improves the ability of the
-Calico components to read from the database (which is their most common
-operation), as they can distribute their reads around the cluster.
+**Main task**: Increases scale by reducing each node’s impact on the datastore. It is one of the {{site.prodname}} [CNI plugins]({{site.baseurl}}/reference/cni-plugin/configuration).
 
-#### Communication
+- **Kubernetes API datastore (kdd)**
 
-etcd is also used as a communication bus between components. We do this
-by having the non-etcd components watch certain points in the keyspace
-to ensure that they see any changes that have been made, allowing them
-to respond to those changes in a timely manner. This allows the act of
-committing the state to the database to cause that state to be programmed
-into the network.
+   The advantages of using the Kubernetes API datastore (kdd) with {{site.prodname}} are:
 
-## BGP client (BIRD)
+   - Simpler to manage because it does not require an extra datastore
+   - Use Kubernetes RBAC to control access to {{site.prodname}} resources
+   - Use Kubernetes audit logging to generate audit logs of changes to {{site.prodname}} resources
 
-{{site.prodname}} deploys a BGP client on every node that also hosts a [Felix](#felix). The role of the BGP client is to read routing state that [Felix](#felix) programs into the kernel and
-distribute it around the data center.
+- **etcd**
 
-The BGP client is responsible for performing the following task:
+   etcd is a consistent, highhly-available distributed key-value store that provides data storage for the {{site.prodname}} network, and forcommunications between components. etcd is supported only for protecting non-cluster hosts (as of {{site.prodname}} v3.1). For completeness, etcd advantages are:
```

Comment:
typo `highhly`
also suggest putting `etcd`  in back ticks

Reply:
Fixed typo, added back ticks.